mangarack-component-core/src/providers/dynastyreader/default.ts

```diff
@@ -0,0 +1,35 @@
+import * as mio from '../../default';
+import {createSeriesAsync} from './series';
+
+/**
+ * Represents the provider.
+ * @internal
+ */
+export const dynastyreader: mio.IProvider = {
+  /**
+   * Contains the name.
+   */
+  name: 'dynastyreader',
+
+  /**
+   * Determines whether the address is a supported address.
+   * @param address The address.
+   * @return Indicates whether the address is a supported address.
+   */
+  isSupported: function(address: string): boolean {
+    return /^http:\/\/dynasty-scans\.com\/series\//.test(address) && !/\/$/.test(address);
+  },
+
+  /**
+   * Promises the series.
+   * @param address The address.
+   * @return The promise for the series.
+   */
+  seriesAsync: function(address: string): Promise<mio.ISeries> {
+    if (!dynastyreader.isSupported(address)) {
+      throw new Error(`Invalid series address: ${address}`);
+    }
+
```

Comment:
This is again a particular weirdness of mine, could you make the return into an `else` statement? Right now, you really have to read the throw statement to see the function can exit there, while an else would make it immediately clear the function can stop there. This merely improves readability, and I particularly love readability.